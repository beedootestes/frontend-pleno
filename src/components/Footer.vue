<template>
  <section class="footer">
    <Link
      to="feed"
      title="Feed"
      class="feed"
      :class="{ active: isActive === 'feed' }"
      @click="activeMenu('feed')">
      <rss-icon class="icon-menu-bar" :size="50" />
    </Link>

    <Link
      to="wiki"
      title="Wiki"
      class="wiki"
      :class="{ active: isActive === 'wiki' }"
      @click="activeMenu('wiki')">
      <light-bulb-outline-icon class="icon-menu-bar" :size="50" />
    </Link>

    <div class="main-link">
      <router-link to="/">
        <img src="../assets/icon-beebot.png" />
      </router-link>
    </div>

    <Link
      to="exam"
      title="Teste relÃ£..."
      class="exam"
      :class="{ active: isActive === 'exam' }"
      @click="activeMenu('exam')">
      <lightning-bolt-outline-icon class="icon-menu-bar" :size="50" />
    </Link>

    <Link
      to="courses"
      title="Cursos"
      class="courses"
      :class="{ active: isActive === 'courses' }"
      @click="activeMenu('courses')">
      <checkbox-marked-outline-icon class="icon-menu-bar" :size="50" />
    </Link>
  </section>
</template>

<script setup lang="ts">
import RssIcon from 'vue-material-design-icons/Rss.vue';
import LightBulbOutlineIcon from 'vue-material-design-icons/LightbulbOutline.vue';
import LightningBoltOutlineIcon from 'vue-material-design-icons/LightningBoltOutline.vue';
import CheckboxMarkedOutlineIcon from 'vue-material-design-icons/CheckboxMarkedOutline.vue';
import Link from "./Link.vue";
import { ref } from "vue";

const isActive = ref<String>('')
const activeMenu = (menu: String): void => { isActive.value = menu }
</script>

<style lang="scss" scoped>
.footer {
  display: flex;
  width: 100%;
  border-top: 1px solid $color-border;
  justify-content: space-between;
  height: 84px;
  box-sizing: border-box;

  .icon-menu-bar {
    margin-bottom: 5px;
  }

  .main-link {
    width: 195px;
    height: 100%;
    background-color: $bg-color-gray;
    border-top-left-radius: 85px 85px;
    border-top-right-radius: 85px 85px;

    img {
      height: 100%;
      display: block;
      margin: auto;
    }
  }
}
</style>