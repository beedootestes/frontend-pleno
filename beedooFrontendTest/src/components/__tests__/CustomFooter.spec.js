import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import Footer from '../CustomFooter.vue'

describe('CustomFooter.vue', () => {
  it('Snapshot do arquivo CustomFooter', () => {
    const wrapper = mount(Footer)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('Testa se o componente Footer é uma instância do Vue', () => {
  it('é uma instancia vue', () => {
    const wrapper = mount(Footer)
    expect(wrapper.vm).toBeTruthy()
  })
})
