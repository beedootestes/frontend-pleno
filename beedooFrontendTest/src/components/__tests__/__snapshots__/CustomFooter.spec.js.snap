// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CustomFooter.vue > Snapshot do arquivo CustomFooter 1`] = `
"<v-layout class=\\"overflow-visible\\" style=\\"height: 56px;\\">
  <v-bottom-navigation color=\\"primary\\" grow=\\"\\">
    <v-btn>
      <v-icon>mdi-rss</v-icon> Feed
    </v-btn>
    <v-btn>
      <v-icon>mdi-lightbulb-on-outline</v-icon> Wiki
    </v-btn>
    <v-btn>
      <v-icon>mdi-flash-outline</v-icon> Teste Rel√¢mpago
    </v-btn>
    <v-btn>
      <v-icon>mdi-checkbox-marked-outline</v-icon> Cursos
    </v-btn>
  </v-bottom-navigation>
</v-layout>"
`;
